### PR TITLE
Yahoo remove editing req

### PR DESCRIPTION
**Pull request by BruceBGordon**
Removes the need to edit new versions of UIAutoMonkey.js for users who don't want the auto-release feature. Still runs automatically for users that use the monkey as is.

Users that are used to commenting out the monkey will need to do a one-time adjustment to set a global variable that controls the auto-release. See the readme for more info.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,32 +37,32 @@ For simplicity's sake, the tool is just a single script you paste into a UI Auto
...
-      },
-
-      // Probability that touch events will have these different properties
-      touchProbability: {
-        multipleTaps: 0.05,
-        multipleTouches: 0.05,
-        longPress: 0.05
-      }
-    },
+	config: {
```
Comment (jonathanpenn) [inline README.md]: Out of curiosity, why was this indented further? Since it doesn't look like the contents significantly changed, can we get rid of the change in whitespace so there's less unnecessary change in this PR?

File: UIAutoMonkey.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ function UIAutoMonkey() {
 		/**
 		* Sometimes the monkey can fall into UI Holes from which it it is hard to escape. The monkey may then spend an inordinate
 		* amount of time in such holes, neglecting other parts of the application.
+
```
Comment (jonathanpenn) [inline UIAutoMonkey.js]: Could this whitespace be removed?

File: UIAutoMonkey.js [hunk 3]
```diff
@@ -120,6 +121,7 @@ function UIAutoMonkey() {
 		UIALogger.logMessage('On Alert: ' + title);
 		return true;
 	}
+
```
Comment (jonathanpenn) [inline UIAutoMonkey.js]: Same removal of whitespace?

File: UIAutoMonkey.js [hunk 4]
```diff
@@ -240,6 +242,7 @@ UIAutoMonkey.prototype.RELEASE_THE_MONKEY = function() {
     } 
 	
 	var targetBundleId = this.target().frontMostApp().bundleID();
+
```
Comment (jonathanpenn) [inline UIAutoMonkey.js]: Third time's a charm? :)

File: README.md [hunk 5]
```diff
@@ -74,30 +74,56 @@ At the top of the script, you'll see a JavaScript dictionary of configuration se
...
+###SetGlobals.js
+Have at least this global there:
+
+```
+UIAutoMonkeyClientWillReleaseTheMonkey = true;
+```
+
+When the monkey is released using the default release mechanism (UIAutoMonkeyClientWillReleaseTheMonkey is undefined or false) it executes
+```UIALogger.logDebug("Releasing the monkey directly from UIAutoMonkey");
+```
```
Comment (jonathanpenn) [inline README.md]: Might want to double check this code snippet. It's rendering funny when I preview the markdown file. This is what I'm seeing:

screen shot 2015-06-28 at 9 43 16 pm

File: README.md [hunk 6]
```diff
@@ -137,6 +163,7 @@ You can add as many handlers as you want. ButtonHander is just a specific type o
 
 Now the conditionHandlers are processed right in the monkey's inner loop, so it is usually prudent to not check the handlers on every single event. That would be inefficient, and furthermore you don't want to jump out of UI Holes immediately: we want to linger for a while. The second parameter is how often (in event units) to check if the buttonHander's condition `isTrue()`. The 3rd parm is `true` if the button descends from the navigation bar, or `false` if it is a top level button.
 
+
```
Comment (jonathanpenn) [inline README.md]: White space?

File: README.md [hunk 7]
```diff
@@ -167,24 +194,25 @@ The monkey can check to see if the application is progressing. It does this by u
...
-        if (monkey.config.anrSettings.debug) {
-            UIALogger.logDebug("fingerprintFunction tree=" + aString);
-        }
-    }
-    catch (e) {
-        aString = "fingerprintFunction error:" + e;
-        UIALogger.logWarning(aString);
-    }
-    return aString;
+	var mainWindow = UIATarget.localTarget().frontMostApp().mainWindow();
```
Comment (jonathanpenn) [inline README.md]: Again, can you double check the white space introduced here? I'd like to reduce the diff of this PR down to necessary changes if possible.

File: README.md (hunk 1 above)
Comment (BruceBGordon) [inline README.md]: done

**Comment by jonathanpenn** [comment]
Nice!
